### PR TITLE
Fix ID cross-references

### LINKED ISSUE #102: Fix ID-based cross-resource references
### What problem are you facing?

Now that https://github.com/crossplane-contrib/terrajet/issues/154 has been resolved, we can fix ID-based cross-resource references previously removed from `provider-jet-azure`.

### PR DESCRIPTION
**Pull request by ulucinar**
Signed-off-by: Alper Rifat Ulucinar <ulucinar@users.noreply.github.com>

### Description of your changes

Fixes #102

I have:

- [X] Read and followed Crossplane's [contribution process].
- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Authored the example manifests for PostgreSQL [flexible server](https://github.com/crossplane-contrib/provider-jet-azure/blob/fbe91f8afeaa9a196fec31512c4923a7aa0e803f/examples/postgresql/flexiblepostgresqlserver.yaml) and [flexible server configuration](https://github.com/crossplane-contrib/provider-jet-azure/blob/fbe91f8afeaa9a196fec31512c4923a7aa0e803f/examples/postgresql/flexiblepostgresqlconfiguration.yaml). Flexible server configuration refers to its associated server via its ID. Tested successfully using these two manifests.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/monitor/v1alpha1/zz_metricalert_terraformed.go [hunk 1]
```diff
@@ -54,6 +54,14 @@ func (tr *MetricAlert) SetObservation(obs map[string]interface{}) error {
 	return json.TFParser.Unmarshal(p, &tr.Status.AtProvider)
 }
 
+// GetID returns ID of underlying Terraform resource of this MetricAlert
```
Comment (sergenyalcin) [inline apis/monitor/v1alpha1/zz_metricalert_terraformed.go:57]: The `MetricAlert` resource was implemented with the following PR: https://github.com/crossplane-contrib/provider-jet-azure/pull/94

While implementing the `MetricAlert` resource, we observed that, it has `ActionGroup` resource id reference. However, as there was no resource id reference support in `Terrajet`, this reference was not configured for this field.

It seems that with this PR, we will now be able to use `ActionGroup` id reference in the `MetricAlert` resource, right?

**Review by sergenyalcin — COMMENTED** [review]
@ulucinar Thank you for this PR

File: apis/monitor/v1alpha1/zz_metricalert_terraformed.go (hunk 1 above)
Comment (ulucinar) [reply apis/monitor/v1alpha1/zz_metricalert_terraformed.go:57]: Yes, that's right. Let's get this merged and we can add/configure/exemplify with a follow-up PR.

**Review by sergenyalcin — APPROVED** [review]
LGTM!

**Pull request merged**